src/bvals/cc/sparse_bvals_cc_in_one.cpp

```diff
@@ -0,0 +1,450 @@
+    md->send_bnd_info_h = Kokkos::create_mirror_view(md->send_bnd_info);
+
+    int iarr = 0;
+    IterateBoundaries(md, [&](sp_mb_t pmb, sp_mbd_t rc, nb_t &nb, const sp_cv_t v) {
+      md->send_bnd_info_h(iarr).allocated = v->IsAllocated();
+      if (v->IsAllocated()) {
+        md->send_bnd_info_h(iarr) = BndInfo::GetSendBndInfo(pmb, nb, v);
+        auto &buf = *md->send_buf_vec[iarr];
+        md->send_bnd_info_h(iarr).buf = buf.buffer();
+      }
+      ++iarr;
+    });
+    Kokkos::deep_copy(md->send_bnd_info, md->send_bnd_info_h);
+  }
+
+  // Restrict
+  {
+    WriteRegion region("restrict");
+    auto &rc = md->GetBlockData(0);
+    auto pmb = rc->GetBlockPointer();
+    IndexShape cellbounds = pmb->cellbounds;
+    IndexShape c_cellbounds = pmb->c_cellbounds;
+    cell_centered_refinement::Restrict(md->send_bnd_info, cellbounds, c_cellbounds);
+  }
+
+  // Load buffer data
+  const Real threshold = Globals::sparse_config.allocation_threshold;
+  auto &bnd_info = md->send_bnd_info;
+  PARTHENON_DEBUG_REQUIRE(bnd_info.size() == nbound, "Need same size for boundary info");
+  ParArray1D<bool> sending_nonzero_flags("sending_nonzero_flags", nbound);
+  auto sending_nonzero_flags_h = Kokkos::create_mirror_view(sending_nonzero_flags);
+  {
+    WriteRegion region("launch buffer load");
+    Kokkos::parallel_for(
+        "SendBoundaryBuffers",
+        Kokkos::TeamPolicy<>(parthenon::DevExecSpace(), nbound, Kokkos::AUTO),
+        KOKKOS_LAMBDA(parthenon::team_mbr_t team_member) {
+          const int b = team_member.league_rank();
+
+          sending_nonzero_flags(b) = false;
+          if (!bnd_info(b).allocated) return;
+
+          const int &si = bnd_info(b).si;
+          const int &ei = bnd_info(b).ei;
+          const int &sj = bnd_info(b).sj;
+          const int &ej = bnd_info(b).ej;
+          const int &sk = bnd_info(b).sk;
+          const int &ek = bnd_info(b).ek;
+          const int Ni = ei + 1 - si;
+          const int Nj = ej + 1 - sj;
+          const int Nk = ek + 1 - sk;
+
+          const int &Nv = bnd_info(b).Nv;
+          const int &Nt = bnd_info(b).Nt;
+          const int &Nu = bnd_info(b).Nu;
+
+          const int NjNi = Nj * Ni;
+          const int NkNjNi = Nk * NjNi;
+          const int NvNkNjNi = Nv * NkNjNi;
+          const int NtNvNkNjNi = Nt * NvNkNjNi;
+          const int NuNtNvNkNjNi = Nu * NtNvNkNjNi;
+
+          Kokkos::parallel_for(
+              Kokkos::TeamThreadRange<>(team_member, NuNtNvNkNjNi), [&](const int idx) {
+                // TODO(LFR): Make sure the tuv indexing is correct, what is currently
+                // here doesn't seem right.
+                const int u = idx / NtNvNkNjNi;
+                const int t = (idx % NtNvNkNjNi) / NvNkNjNi;
+                const int v = (idx % NvNkNjNi) / NkNjNi;
+                const int k = (idx % NkNjNi) / NjNi + sk;
+                const int j = (idx % NjNi) / Ni + sj;
+                const int i = idx % Ni + si;
```

Comment:
Is there a reason the ordering is now `vtu` here? Elsewhere I think it was always `tuv`. This indexing is different than what I used but it seems to work for e.g. `Nu = 3 Nt = 2 Nv = 1 Nk = 1 Nj = 1 Ni = 1`:
`idx: 0 1 2 3 4 5`
`u   : 0 0 1 1 2 2 `
`t   : 0 1 0 1 0 1`
else `0`

Reply:
I need to go back and look, but I was concerned I had copied and pasted this from somewhere. If I just wrote it and forget the order of the alphabet, it should just be changed.

Reply:
After looking around, I think the ordering was just wrong. I believe this should have resulted in the wrong memory being written for arrays with `Nu != 1` or `Nt != 1`, but this probably would not get picked up in Parthenon tests. It has been fixed now.